internal/pkg/coordinator/monitor.go

```diff
@@ -0,0 +1,360 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package coordinator
+
+import (
+	"context"
+	"encoding/json"
+	"net"
+	"os"
+	"runtime"
+	"sync"
+	"time"
+
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+
+	"fleet/internal/pkg/bulk"
+	"fleet/internal/pkg/config"
+	"fleet/internal/pkg/dl"
+	"fleet/internal/pkg/es"
+	"fleet/internal/pkg/model"
+	"fleet/internal/pkg/monitor"
+)
+
+const (
+	defaultCheckInterval           = 20 * time.Second // check for valid leaders every 20 seconds
```

Comment:
Making this configurable later on will become useful for integration tests. Then short intervals can be used with multiple fleet-servers to see if things work as expected.

Reply:
Yep, it is exposed as a value on the struct allowing it to be adjusted in integration test before starting the monitor.